src/Bicep.LangServer/Handlers/BicepHoverHandler.cs

```diff
@@ -31,11 +33,24 @@ public BicepHoverHandler(ISymbolResolver symbolResolver)
         public override Task<Hover?> Handle(HoverParams request, CancellationToken cancellationToken)
         {
             var result = this.symbolResolver.ResolveSymbol(request.TextDocument.Uri, request.Position);
+
             if (result == null)
             {
                 return Task.FromResult<Hover?>(null);
             }
 
+            // We will not provide hover information on keywords
+            if (result.Origin is ITopLevelNamedDeclarationSyntax topLevelNamedDeclarationSyntax &&
+                topLevelNamedDeclarationSyntax.Keyword is Token keyword)
+            {
+                var position = PositionHelper.GetOffset(result.Context.LineStarts, request.Position);
+
+                if (keyword.Span.ContainsInclusive(position))
+                {
+                    return Task.FromResult<Hover?>(null);
+                }
+            }
```

Comment:
This feels too specific and will leave other edge cases open - for example, hovering on the `=` of a resource declaration syntax will have the same issue.

The more general problem seems to be that the hover `Range` below can be outside the position in `request.Position`. Can't we just calculate `PositionHelper.GetNameRange(result.Context.LineStarts, result.Origin)` first and return `null` if it doesn't contain `request.Position`?